### PR TITLE
more generic, but still very basic, ansible task examples.

### PR DESCRIPTION
**Pull request by konstruktoid**
Signed-off-by: Thomas Sjögren <konstruktoid@users.noreply.github.com>

### CODE REVIEW

**Comment by illfated** [comment]
#sign-off

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md [hunk 1]
```diff
@@ -36,14 +36,13 @@ This topic describes how to deploy Microsoft Defender ATP for Linux using Ansibl
 Before you get started, please see [the main Microsoft Defender ATP for Linux page](microsoft-defender-atp-linux.md) for a description of prerequisites and system requirements for the current software version.
 
 - Ansible needs to be installed on at least on one computer (we will call it the master).
-- Passwordless SSH must be configured for the root user between the master and all clients.
+- SSH must be configured for an administrator account between the master and all clients.
```
Comment (nawate) [inline windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: Hi Thomas, Thanks for taking time and making changes!

Here the intention was to let the Administrator know that the SSH must be configured "passwordless" between master and clients. 
If we just say "configured" it's not intuitive.

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md [hunk 2]
```diff
@@ -36,14 +36,13 @@ This topic describes how to deploy Microsoft Defender ATP for Linux using Ansibl
...
 - The following software must be installed on all clients:
-  - Python-apt
```
Comment (nawate) [inline windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: python-apt is a necessary module to run apt commands using ansible. When I tried w/o installing python-apt it simply failed.
Pls refer https://docs.ansible.com/ansible/latest/modules/apt_module.html

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md [hunk 3]
```diff
@@ -84,37 +83,39 @@ Create subtask or role files that contribute to an actual task. Create the follo
 - Copy the onboarding package to all client machines:
 
     ```bash
-    $ cat /etc/ansible/roles/copy_onboarding_pkg.yml
```
Comment (nawate) [inline windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: I'm fine with removing cat command; Here intention was to let the Admin guy know that the file should look like this.
Can we replace cat -> vi

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md [hunk 4]
```diff
@@ -84,37 +83,39 @@ Create subtask or role files that contribute to an actual task. Create the follo
...
+      apt_key:
+        url: https://packages.microsoft.com/keys/microsoft.asc
+        state: present
+      when: ansible_os_family == "Debian"
     ```
 
 - Create a `setup.sh` script that operates on the onboarding file:
 
     ```bash
-    $ cat /root/setup.sh
```
Comment (nawate) [inline windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: Here we are removing it. But we are using it on line #121 with as /root/setup.sh
This line tells admin that create a file under "/root" directory.

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md (hunk 1 above)
Comment (konstruktoid) [reply windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: Hi @nawate, SSH doesn't actually need to be "passwordless" but it's recommended (so the playbooks can run smoothly) to use public key authentication, and then use `become: 'yes'` and `become_method: sudo` to escalate privileges for the administrative account.
But is it enough to change the wording to the following?
```
- SSH must be configured for an administrator account between the master and all clients, and it is recommended be configured with public key authentication.
```
Then we leave it to the reader to distribute the public key(s) to the clients.

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md (hunk 3 above)
Comment (konstruktoid) [reply windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: We've already stated that the file should be created in the `/etc/ansible/roles/` directory, but will update the sentence.

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md (hunk 4 above)
Comment (konstruktoid) [reply windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: Will update the sentence, and ignore the fact that all examples is using the `/root` directory and the `root` user. Probably a later update :)

File: windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md (hunk 2 above)
Comment (konstruktoid) [reply windows/security/threat-protection/microsoft-defender-atp/linux-install-with-ansible.md]: You're correct, will fix.
Ansible output:
```
 [WARNING]: Updating cache and auto-installing missing dependency: python-apt
```

**Comment by e0i** [comment]
@konstruktoid Would you be kind to resolve the merge conflicts within the PR? Thank you.

**Comment by konstruktoid** [comment]
@e0i, fixed.

**Comment by illfated** [comment]
BTW, was this PR based on any issue ticket feedback?

**Comment by konstruktoid** [comment]
@illfated, nope.

**Comment by illfated** [comment]
#sign-off

**Comment by e0i** [comment]
@mjcaparas Looks like the PR is ready to merge. Thanks.

**Pull request merged**